File: rust/a2022/src/bin/01.rs
use std::cmp::Reverse;
use std::collections::BinaryHeap;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut heap = BinaryHeap::from([Reverse(0); 3]);
    let mut acc: i32 = 0;

    for line in common::get_problem_lines(2022, 1) {
        if line.is_empty() {
            heap.push(Reverse(acc));
            heap.pop();
            acc = 0;
        } else {
            acc += line.parse::<i32>()?;
        }
    }
    heap.push(Reverse(acc));
    heap.pop();
    let v: Vec<i32> = heap.into_sorted_vec().into_iter().map(|a| a.0).collect();
    println!("{:?}", v);
    println!("{:?}", v.iter().sum::<i32>());

    Ok(())
}

File: python/a2020/01.py
from common import get_problem_lines
input = [int(i) for i in get_problem_lines()]
print(f"part 1: {next((i*(2020-i) for i in input if 2020 - i in input))}")
print(f"part 2: {next((i*j*(2020-i-j) for i in input for j in input if 2020 - i - j in input))}")
